services/_internal/volume-stage-in/main.py

```diff
@@ -81,18 +79,27 @@ def list_b2share_record_files(url):
 
 
 def download_file(url, local_filename=None):
-    if not local_filename:
-        local_filename = url.split('/')[-1]
     r = requests.get(url, stream=True, verify=False)
-    with open(os.path.join(download_dir, local_filename), 'wb') as f:
+    disp = r.headers['content-disposition']
+    fname = re.findall("filename=(.+)", disp)
```

Comment:
I'm not sure if any letters in the headers can be capital, but in this case "re" is case-sensitive.